server/services/scraperAmazon.js

```diff
@@ -0,0 +1,30 @@
+const puppeteer = require("puppeteer");
+
+function scrapeAmazon(url) {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const browser = await puppeteer.launch();
```

Comment:
Why are you using both `new Promise` and `async/await` syntaxes?

Reply:
Because the puppeteer methods always return a promise so that the logic in the new Promise is async. It could be either an async/await pattern or using the .then method. I chose async/await as it looks cleaner.